UI/js/signup.js

```diff
@@ -0,0 +1,267 @@
+  password = document.getElementById('register-password'),
+  passwordErrorMessage = document.getElementById('register-password-error'),
+  confirmPassword = document.getElementById('register-confirm-password'),
+  confirmPasswordErrorMessage = document.getElementById('register-confirm-password-error'),
+  signupButton = document.querySelector('#signup-form-btn');
+
+// regex references
+// https://stackoverflow.com/questions/20003870/how-do-i-create-regex-of-min-8-max-16-alphabetic-numbers-and-no-space
+// https://stackoverflow.com/questions/201323/how-to-validate-an-email-address-using-a-regular-expression
+// https://stackoverflow.com/questions/9289451/regular-expression-for-alphabets-with-spaces
+
+const errors = {};
+
+signupButton.addEventListener('mouseenter', () => {
+  if (!firstName.value.trim()) {
+    errors.firstName = 'First name is required';
+    firstNameErrorMessage.classList.add('is-visible');
+    firstNameErrorMessage.innerHTML = errors.firstName;
+  } else if (firstName.value.trim() && !(/^[A-Za-z]*$/).test(firstName.value.trim())) {
+    errors.firstName = `First name must consist of only alphabets
+    and must contain no spaces between characters`;
+    firstNameErrorMessage.classList.add('is-visible');
+    firstNameErrorMessage.innerHTML = errors.firstName;
+  } else if (firstName.value.trim() && firstName.value.trim().length > 50) {
+    errors.firstName = 'First name must be fewer than 50 characters';
+    firstNameErrorMessage.classList.add('is-visible');
+    firstNameErrorMessage.innerHTML = errors.firstName;
+  } else {
+    delete (errors.firstName);
+  }
+
+
+  if (!lastName.value.trim()) {
+    errors.lastName = 'Last name is required';
+    lastNameErrorMessage.classList.add('is-visible');
+    lastNameErrorMessage.innerHTML = errors.lastName;
+  } else if (lastName.value.trim() && !/^[A-Za-z]*$/.test(lastName.value.trim())) {
+    errors.lastName = `Last name must consist of only alphabets
+    and must contain no spaces between characters`;
+    lastNameErrorMessage.classList.add('is-visible');
+    lastNameErrorMessage.innerHTML = errors.lastName;
+  } else if (lastName.value.trim() && lastName.value.trim().length > 50) {
+    errors.lastName = 'Last name must be fewer than 50 characters';
+    lastNameErrorMessage.classList.add('is-visible');
+    lastNameErrorMessage.innerHTML = errors.lastName;
+  } else {
+    delete (errors.lastName);
+  }
+
+
+  if (!email.value.trim()) {
+    errors.email = 'Email is required';
+    emailErrorMessage.classList.add('is-visible');
+    emailErrorMessage.innerHTML = errors.email;
+  } else if (email.value.trim() && (!(/^[^\s@]+@([^\s@.,]+\.)+[^\s@.,]{2,}$/).test(email.value.trim()))) {
+    errors.email = 'Email is invalid';
+    emailErrorMessage.classList.add('is-visible');
+    emailErrorMessage.innerHTML = errors.email;
+  } else {
+    delete (errors.email);
+  }
+
+  if (!password.value.trim()) {
```

Comment:
Password and confirmPassword fields can also be validated by the same function. I think this will reduce the amount of code you'd have to write.